CrossrefPlugin.php

```diff
@@ -411,4 +416,96 @@ public function getAllowedDoiTypes(): array
     }
+
+    /**
+     * Make additional validation checks against publishing requirements
+     *
+     * @param $hookName string
+     * @param $args array [
+     * @return bool
+     * @option array Validation errors already identified
+     * @option Publication The publication to validate
+     * @option Submission The submission of the publication being validated
+     * @option array The locales accepted for this object
+     * @option string The primary locale for this object
+     * ]
+     * @see PKPPublicationService::validatePublish()
+     */
+    public function validate(string $hookName, array $args): bool
+    {
+        $errors =& $args[0];
+        $submission = $args[2];
+        $context = Application::getContextDAO()->getById($submission->getData('contextId'));
+        $publication = $submission->getCurrentPublication();
+        $issueId = $publication->getData('issueId');
+        $primaryLocale = $context->getPrimaryLocale();
+        $rules = [
+            'printIssn' => ['required', 'string'],
+            'onlineIssn' => ['required', 'string'],
+            'title' => ['required', 'array'],
+            'title.' . $primaryLocale => ['required', 'string'],
+            'doi' => ['required', 'url'],
+            'journalTitle' => ['required', 'array'],
+            'journalTitle.' . $primaryLocale => ['required', 'string'],
+            'issueId' => ['required', 'integer'],
+            'dateSubmitted' => ['required', 'date'],
+        ];
+        $metadata = [
+            'title'=> $publication->getData('title'),
+            'printIssn'=> $context->getData('printIssn'),
+            'onlineIssn'=> $context->getData('onlineIssn'),
+            'doi'=> $publication->getDoi(),
+            'dateSubmitted'=> $submission->getData('dateSubmitted'),
+            'journalTitle'=> $context->getData('name'),
+            'issueId'=> $issueId,
+        ];
+
+        $validator = Validator::make(
+            $metadata,
+            $rules,
+            $this->getValidationMessages($primaryLocale)
+        );
+        if(!$validator->passes()){
+            $errors = $this->formatErrors($validator->errors()->toArray());
+        }
+        return HOOK::CONTINUE;
+    }
+
+    /**
+     * @return array
+     */
+    private function getValidationMessages($primaryLocale): array
+    {
+        return [
+            'issueId.required' => __('plugins.generic.crossref.issueId.required'),
+            'doi.required' => __('plugins.generic.crossref.doi.required'),
+            'doi.url' => __('plugins.generic.crossref.doi.url'),
+            'onlineIssn.required' => __('plugins.generic.crossref.onlineIssn.required'),
+            'printIssn.required' => __('plugins.generic.crossref.printIssn.required'),
+            'dateSubmitted.required' => __('plugins.generic.crossref.dateSubmitted.required'),
+            'dateSubmitted.date' => __('plugins.generic.crossref.dateSubmitted.date'),
```

Comment:
The submission date comes from a SQL DATETIME column, so it will always be valid. This rule (and the locale key) should be removed.